panic / core dump when viewing details on shallow commits

**Describe the bug**
```
panic: Any { .. }
trace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: start_thread
   7: __GI___clone

Aborted (core dumped)
```
**To Reproduce**
`git clone --depth 1 https://github.com/extrawurst/gitui` (note the `--depth 1` flag)
./gitui -> go to the log tab -> press enter to view details on the shallow commit

**Expected behavior**
Show details of the commit

**Context (please complete the following information):**
OS: Arch Linux
Kernel: x86_64 Linux 5.13.9-arch1-1
GitUI Version: 0.16.2
Rust version: rustc 1.54.0

**Additional context**
https://git-scm.com/docs/shallow

Useful, for example, when cloning plugins for vim (since only the last commit is fetched, it's faster that way)

the actual error:
```
[ERROR] panic: PanicInfo { payload: Any { .. }, message: Some(failed to fetch: Git(Error { code: -3, klass: 9, message: "object not found - no match for id (d6f43f063eedf95e33d019f41265e1d377a572ec)" })), location: Location { f
ile: "asyncgit/src/commit_files.rs", line: 72, col: 18 } }
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Fixed
 - do not allow to ignore .gitignore files ([#825](https://github.com/extrawurst/gitui/issues/825))
+- crash in shallow repo ([#836](https://github.com/extrawurst/gitui/issues/836))
 
 ## [0.16.2] - 2021-07-10
 

--- a/asyncgit/src/sync/commit_files.rs
+++ b/asyncgit/src/sync/commit_files.rs
@@ -49,7 +49,9 @@ pub(crate) fn get_commit_diff(
     let commit = repo.find_commit(id.into())?;
     let commit_tree = commit.tree()?;
     let parent = if commit.parent_count() > 0 {
-        Some(repo.find_commit(commit.parent_id(0)?)?.tree()?)
+        repo.find_commit(commit.parent_id(0)?)
+            .ok()
+            .and_then(|c| c.tree().ok())
     } else {
         None
     };

